File: Assets/Scripts/Networking/DTO/PuzzleMetadataDTO.cs
using System;
using PuzzleGeneration;

namespace Networking.DTO
{
    [Serializable]
    public class PuzzleMetadataDTO
    {
        public string onlineID;
        public string name;
        public string authorId;
        public string author;
        public PuzzleLayout layout;
        public ContentDTO content;
    }
}

File: PuzzleServer/src/main/kotlin/com/vrpuzzle/puzzleserver/model/dto/PuzzleMetadataDTO.kt
package com.vrpuzzle.puzzleserver.model.dto

import com.vrpuzzle.puzzleserver.model.type.PuzzleLayout

data class PuzzleMetadataDTO(
    val onlineID: String,
    val name: String,
    val authorId: String,
    val author: String,
    val layout: PuzzleLayout,
    val content: ContentDTO
)